### PR TITLE
Feature: Batch Insights support

### PR DESCRIPTION
**Pull request by brnleehng**
#335

### CODE REVIEW

File: R/batch-api.R [hunk 1]
```diff
@@ -235,8 +235,15 @@ BatchUtilities <- R6::R6Class(
           commands <- c(commands, args$commandLine)
         }
 
+        commands <- linuxWrapCommands(commands)
+        if (!is.null(args$applicationInsights)) {
+          commands <- gsub("wait",
```
Comment (bgklein) [inline R/batch-api.R]: Is wait something you add to the user cmd?

Comment (bgklein) [reply R/batch-api.R]: I'm assuming so

File: docs/90-troubleshooting.md [hunk 2]
```diff
@@ -2,6 +2,22 @@
...
+``` json
+"applicationInsights": {
+    "applicationId": "",
+    "instrumentationKey": ""
+  }
+```
+Note: Github has a rate limit of up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address. [Link](https://developer.github.com/v3/#rate-limiting)
+
+For more information about these tools,
+- [Batch Explorer](https://github.com/azure/batchexplorer)
```
Comment (bgklein) [inline docs/90-troubleshooting.md]: https://azure.github.io/BatchExplorer/ ?

File: inst/startup/cluster_setup.sh [hunk 3]
```diff
@@ -1,30 +1,4 @@
 #!/bin/bash
-
-# Entry point for the start task. It will install the docker runtime and pull down the required docker images
-# Usage:
-# setup_node.sh
-
-apt-get -y install linux-image-extra-$(uname -r) linux-image-extra-virtual
```
Comment (bgklein) [inline inst/startup/cluster_setup.sh]: Reason for this being removed? Just unneeded now? I don't see it  moved to code anywhere.

**Comment by sorenvind** [comment]
@brnleehng The merge of this PR is the root cause of the bug I attempted to fix in https://github.com/Azure/doAzureParallel/pull/340. However, the bug fix did not succeed.

By merging the present PR to master, the file at `https://raw.githubusercontent.com/Azure/doAzureParallel/master/inst/startup/cluster_setup.sh` was changed. The file is referenced in old versions of doAzureParallel (see e.g. https://github.com/Azure/doAzureParallel/blob/v0.6.3/R/cluster.R#L145). Since the set up of docker was removed by this PR, all pools/jobs orchestrated using previous versions of doAzureParallel will fail until this PR is reverted or docker is installed again by the cluster_setup.sh file. It is unclear to me if the version on master could potentially work.

This change has broken the daily runs in our production system relying on Batch for two days now. What can I do to help fix this?

**Pull request merged**